<?xml version="1.0" ?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
<xacro:macro name="astra_rover_mechdesc" params = "">

	<xacro:include filename="$(find astra_rover_description)/urdf/astra_rover_transmission.xacro" />
	

	<!-- * * * Link Definitions * * * -->

	<link name="chassis">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<box size="0.6 0.5 0.2"/>
			</geometry>
			<material name="c">
				<color rgba="1 1 1 1"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="8"/>
			<inertia ixx="0.19"  ixy="0"  ixz="0" iyy="0.19" iyz="0" izz="0.19" />
		</inertial>	

		<collision>
			<geometry>
				<box size="0.6 0.5 0.2"/>
			</geometry>
		</collision>


	</link>

	<link name="rocker_left_front">
		<visual>
			<origin xyz="0.08 0.3 -0.1" rpy="0 1.83 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.3"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>	

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="1.01"/>
			<inertia ixx="0.03"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.03" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.3"/>
			</geometry>
		</collision>


	</link>
	<link name="rocker_left_back">
		<visual>
			<origin xyz="-0.25 0.3 -0.11" rpy="0 -1.83 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.4"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="1.35"/>
			<inertia ixx="0.07"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.07" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.4"/>
			</geometry>
		</collision>	
	</link>
	<link name="bogey_left_front">
		<visual>
			<origin xyz="0.3 0.3 -0.14" rpy="0 1.57 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.2"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>	


		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.67"/>
			<inertia ixx="0.01"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.01" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.2"/>
			</geometry>
		</collision>	

	</link>
	<link name="bogey_left_back">
		<visual>
			<origin xyz="0.12 0.3 -0.2" rpy="0 -2.09 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.24"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>	

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.81"/>
			<inertia ixx="0.015"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.015" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.24"/>
			</geometry>
		</collision>	


	</link>
	<link name="bogey_steering_left">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.50"/>
			<inertia ixx="0.001"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
		</collision>	

	</link>
	<link name="rocker_steering_left">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.50"/>
			<inertia ixx="0.001"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
		</collision>	
	</link>

	<link name="rocker_right_front">
		<visual>
			<origin xyz="0.08 -0.3 -0.1" rpy="0 1.83 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.3"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="1.01"/>
			<inertia ixx="0.03"  ixy="0"  ixz="0" iyy="0.03" iyz="0" izz="0.03" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.3"/>
			</geometry>
		</collision>
	</link>
	<link name="rocker_right_back">
		<visual>
			<origin xyz="-0.25 -0.3 -0.11" rpy="0 -1.83 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.4"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="1.35"/>
			<inertia ixx="0.07"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.07" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.4"/>
			</geometry>
		</collision>	
	</link>
	<link name="bogey_right_front">
		<visual>
			<origin xyz="0.3 -0.3 -0.14" rpy="0 1.57 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.2"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.67"/>
			<inertia ixx="0.01"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.01" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.2"/>
			</geometry>
		</collision>	

	</link>
	<link name="bogey_right_back">
		<visual>
			<origin xyz="0.12 -0.3 -0.2" rpy="0 -2.09 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.24"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>


		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.81"/>
			<inertia ixx="0.015"  ixy="0"  ixz="0" iyy="0.0" iyz="0" izz="0.015" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.24"/>
			</geometry>
		</collision>	



	</link>
	<link name="bogey_steering_right">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>


		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.50"/>
			<inertia ixx="0.001"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
		</collision>	

	</link>
	<link name="rocker_steering_right">
		<visual>
			<origin xyz="0 0 0" rpy="0 0 1.57"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
			<material name="tyre">
				<color rgba="0 0 0 1.0"/>
			</material>
		</visual>


		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.50"/>
			<inertia ixx="0.001"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.02" length="0.15"/>
			</geometry>
		</collision>	


	</link>
	<link name="wheel_right_3">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	


	</link>
	<link name="wheel_right_2">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	



	</link>
	<link name="wheel_right_1">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	


	</link>
	<link name="wheel_left_3">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>
		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	


	</link>
	<link name="wheel_left_2">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	



	</link>
	<link name="wheel_left_1">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="grey1">
				<color rgba="0.9 0.9 0.9 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.62"/>
			<inertia ixx="0.002"  ixy="0"  ixz="0" iyy="0.003" iyz="0" izz="0.002" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
		</collision>	



	</link>
	<link name="rocker_hinge_left">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
			<material name="red">
				<color rgba="1 0.25 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.01"/>
			<inertia ixx="0.0"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
		</collision>	

	</link>

	<link name="rocker_hinge_right">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
			<material name="red">
				<color rgba="1 0.25 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.01"/>
			<inertia ixx="0.0"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
		</collision>	


	</link>
	<link name="bogey_hinge_left">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
			<material name="red">
				<color rgba="1 0.25 0 1.0"/>
			</material>
		</visual>

		<inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.01"/>
			<inertia ixx="0.0"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
		</collision>

		

	</link>

	<link name="bogey_hinge_right">
		<visual>
			<origin xyz="0 0 0" rpy="1.57 0 0"/>
			<axis xyz="0 0 0" />
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
			<material name="red">
				<color rgba="1 0.25 0 1.0"/>
			</material>
		</visual>
        
        <inertial>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<mass value="0.01"/>
			<inertia ixx="0.0"  ixy="0"  ixz="0" iyy="0.001" iyz="0" izz="0.0" />
		</inertial>	

		<collision>
			<geometry>
				<cylinder radius="0.05" length="0.1"/>
			</geometry>
		</collision>	
	</link>


	<!-- * * * Joint Definitions * * * -->


	<joint name="rocker_hinge_right" type="revolute">
		<parent link="chassis"/>
		<child link="rocker_hinge_right"/>
		<origin xyz="-0.06 -0.3 -0.06" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit lower="-3.1416" upper="3.1416" effort="50" velocity="1"/>
	</joint>


	<joint name="rhr" type="fixed">
		<parent link="rocker_hinge_right"/>
		<child link="rocker_right_back"/>
		<origin xyz="0.06 0.3 0.06" rpy="0 0 0"/>
	</joint>

	<joint name="rocker_right" type="fixed">
		<parent link="rocker_right_back"/>
		<child link="rocker_right_front"/>
		<origin xyz="0 0 0" rpy="0 0 0"/>
	</joint>

	<joint name="rocker_hinge_left" type="revolute">
		<parent link="chassis"/>
		<child link="rocker_hinge_left"/>
		<origin xyz="-0.06 0.3 -0.06" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit lower="-3.1416" upper="3.1416" effort="50" velocity="1"/>
	</joint>


	<joint name="rhl" type="fixed">
		<parent link="rocker_hinge_left"/>
		<child link="rocker_left_back"/>
		<origin xyz="0.06 -0.3 0.06" rpy="0 0 0"/>
	</joint>

	<joint name="rocker_left" type="fixed">
		<parent link="rocker_left_back"/>
		<child link="rocker_left_front"/>
		<origin xyz="0 0 0" rpy="0 0 0"/>
	</joint>

	<joint name="bogey_hinge_left" type="revolute">
		<parent link="rocker_left_front"/>
		<child link="bogey_hinge_left"/>
		<origin xyz="0.2 0.3 -0.14" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit lower="-3.1416" upper="3.1416" effort="50" velocity="1"/>
	</joint>

	<joint name="bhl" type="fixed">
		<parent link="bogey_hinge_left"/>
		<child link="bogey_left_back"/>
		<origin xyz="-0.2 -0.3 0.14" rpy="0 0 0"/>
	</joint>

	<joint name="bogey_left" type="fixed">
		<parent link="bogey_left_back"/>
		<child link="bogey_left_front"/>
		<origin xyz="0 0 0" rpy="0 0 0"/>
	</joint>

	<joint name="bogey_hinge_right" type="revolute">
		<parent link="rocker_right_front"/>
		<child link="bogey_hinge_right"/>
		<origin xyz="0.2 -0.3 -0.14" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit lower="-3.1416" upper="3.1416" effort="50" velocity="1"/>
	</joint>

	<joint name="bhr" type="fixed">
		<parent link="bogey_hinge_right"/>
		<child link="bogey_right_back"/>
		<origin xyz="-0.2 0.3 0.14" rpy="0 0 0"/>
	</joint>

	<joint name="bogey_right" type="fixed">
		<parent link="bogey_right_back"/>
		<child link="bogey_right_front"/>
		<origin xyz="0 0 0" rpy="0 0 0"/>
	</joint>


	<!--* * * Wheel Joint Definitions * * *-->

	<joint name="wheel_left_3_joint" type="continuous">
		<parent link="rocker_steering_left"/>
		<child link="wheel_left_3"/>
		<origin xyz="0 0.08 -0.075" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<joint name="wheel_left_2_joint" type="continuous">
		<parent link="bogey_left_back"/>
		<child link="wheel_left_2"/>
		<origin xyz="0.016 0.38 -0.275" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<joint name="wheel_left_1_joint" type="continuous">
		<parent link="bogey_steering_left"/>
		<child link="wheel_left_1"/>
		<origin xyz="0 0.08 -0.075" rpy="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<joint name="wheel_right_3_joint" type="continuous">
		<parent link="rocker_steering_right"/>
		<child link="wheel_right_3"/>
		<origin xyz="0 -0.08 -0.075" rpy="0 0 3.14"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<joint name="wheel_right_2_joint" type="continuous">
		<parent link="bogey_right_back"/>
		<child link="wheel_right_2"/>
		<origin xyz="0.016 -0.38 -0.275" rpy="0 0 3.14"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<joint name="wheel_right_1_joint" type="continuous">
		<parent link="bogey_steering_right"/>
		<child link="wheel_right_1"/>
		<origin xyz="0 -0.08 -0.075" rpy="0 0 3.14"/>
		<axis xyz="0 1 0"/>
		<limit  effort="10" velocity="5"/>
	</joint>

	<!--* * * Steering Joint Definitions * * *-->

	<joint name= "bogey_steering_right_joint" type="revolute">
		<parent link="bogey_right_front"/>
		<child link="bogey_steering_right"/>
		<origin xyz="0.39 -0.3 -0.2" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit lower="-2.356" upper="2.356" effort="15"                     velocity="3"/>
	</joint>

	<joint name="rocker_steering_right_joint" type="revolute">
		<parent link="rocker_right_back"/>
		<child link="rocker_steering_right"/>
		<origin xyz="-0.44 -0.3 -0.2" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit lower="-2.356" upper="2.356" effort="15"                     velocity="3"/>
	</joint>
	<joint name="bogey_steering_left_joint" type="revolute">
		<parent link="bogey_left_front"/>
		<child link="bogey_steering_left"/>
		<origin xyz="0.39 0.3 -0.2" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit lower="-2.356" upper="2.356" effort="15"                     velocity="3"/>
	</joint>

	<joint name="rocker_steering_left_joint" type="revolute">
		<parent link="rocker_left_back"/>
		<child link="rocker_steering_left"/>
		<origin xyz="-0.44 0.3 -0.2" rpy="0 0 0"/>
		<axis xyz="0 0 1"/>
		<limit lower="-2.356" upper="2.356" effort="15"                     velocity="3"/>
	</joint>


	<wheel_transmission wheel_name="wheel_right_1"/>
	<wheel_transmission wheel_name="wheel_right_2"/>
	<wheel_transmission wheel_name="wheel_right_3"/>

	<wheel_transmission wheel_name="wheel_left_1"/>
	<wheel_transmission wheel_name="wheel_left_2"/>
	<wheel_transmission wheel_name="wheel_left_3"/>
	
	<steering_transmission steering_name="bogey_steering_right"/>
	<steering_transmission steering_name="bogey_steering_leftt"/>
	<steering_transmission steering_name="rocker_steering_right"/>
	<steering_transmission steering_name="rocker_steering_left"/>

	<hinge_transmission hinge_name="bogey_hinge_right"/>
	<hinge_transmission hinge_name="bogey_hinge_leftt"/>
	<hinge_transmission hinge_name="rocker_hinge_right"/>
	<hinge_transmission hinge_name="rocker_hinge_left"/>

</xacro:macro>
</robot>
